### ISSUE TITLE
Admin panel: better handling of default values (show what they are, and link to interface to modify)

### ISSUE THREAD

**Issue by marclaporte** [virtualmin/virtualmin-gpl#555]
Virtualmin provides a lot of flexibility and this is great. So for the setting to "Display error messages", we have options for "Default/Yes/No".

It would be nice if it indicated what is current default. And even nicer to have link to the interface to modify it (if user has permissions).
![2023-04-27_165546](https://user-images.githubusercontent.com/1004261/234993692-d855d86b-bde6-4400-aec6-317f4fbacffb.png)

Thank you!

**Comment by marclaporte** [comment]
More questions about default values
![2023-04-27_172919](https://user-images.githubusercontent.com/1004261/234998638-f329b174-cb4b-41a0-b116-1e1e729b9cd9.png)

**Comment by jcameron** [comment]
Sometimes we don't know the defaults, sorry! They are under the control of PHP in cases like this, and may differ depending on the PHP version..

**Comment by chris001** [comment]
[display_errors](https://www.php.net/manual/en/errorfunc.configuration.php#ini.display-errors)

[error_reporting](https://www.php.net/manual/en/function.error-reporting.php)

**Comment by kroky** [comment]
Assuming we have the path to the php executable since we are in a virtualmin domain and the names of the config variables since we are saving them to the php conf file, can't we use one of the following ways to get the default:
`php -r 'echo ini_get("error_reporting");'` (in case the default has not been changed by the virtualmin settings)
`php --info | grep error_reporting` - second value is the master value

Maybe it's worth showing the current values only for settings that are set to use default - this way user will know what would be the default value?

**Comment by jcameron** [comment]
Won't a command like `php -r 'echo ini_get("error_reporting");'` still show the value from the `php.ini` file, not the real default?

**Comment by kroky** [comment]
Yes, it will actually show the currently active setting which could come from the system php.ini or other ini files loaded in addition to the system one. I think the user's problem here is that they are unaware of the default value when php is running a script without any modifications to the settings from the script itself. Displaying the currently active setting in that case is just fine. Of course, it only makes sense if the virtualmin domain settings are set to default. Otherwise, it will display the setting selected in the domain config itself.

**Comment by marclaporte** [comment]
Closely related issue so I opted to keep in same issue:
![image](https://github.com/virtualmin/virtualmin-gpl/assets/1004261/f3372fdf-0ab8-4b59-ba2b-b02230257ae9)

**Comment by jcameron** [comment]
That's a good point, we should always allow switching back to the default (even if we don't know what it is)

**Comment by marclaporte** [comment]
I think the first part of this issue is not resolved yet.

**Comment by marclaporte** [comment]
> It would probably make the page look crowded.

ok, maybe it could live in a pop up or something.


> Do you mean to open it in the editor to edit manually?

A link to where to edit this value in Webmin/Virtualmin. And this link should only appear to users with sufficient permissions.

> add a link to PHP manual for each option, so there a user would be in no time see all defaults and etc

I like that idea for general information. The PHP manual will provide guidance on what each setting is. However, it won't reliably tell me the default setting of my operating system or the current default of my server.

When configuring a Virtualmin Virtual Server, users will benefit to know the default values of the current server (which we should be able to show here) which may have been modified compared to the default values of the operating system (which we don't easily know, so we don't try to do this as discuss at https://github.com/virtualmin/virtualmin-gpl/issues/580) 

Thanks!

**Comment by nekohayo** [comment]
> >  It would probably make the page look crowded.
> 
> ok, maybe it could live in a pop up or something.

You could show a tooltip when the mouse hovers the "( ) Default" option, with the tooltip showing the default value (if known). Either a native browser tooltip, or reusing the kind of tooltip widget you've shown in that screenshot above when hovering the "(?)". That way, it's even less intrusive than the currently proposed approach, though they could be complementary too: you could still have a (?) button linking to the PHP (or any) documentation in addition to the tooltip over the "Default" radiobutton.

**Comment by jcameron** [comment]
I do like the tooltip idea! The tricky part is how to find the default valuer programatically ..

**Comment by iliajie** [comment]
Do you like the idea with the tooltip I suggested earlier (in the screenshot above, to open the link to PHP documentation) or something else? If something else, can you be more specific about what do you think would be useful?

>  The tricky part is how to find the default valuer programatically

I don't believe this is anyhow useful. We already going to show **Default** radio, meaning the option isn't set on the config and default is used. Anyone who needs to find out the actual default value (I don't know why and for what practical purpose) could simply call `phpinfo()` function ..

**Comment by jcameron** [comment]
Linking to the docs would also be useful, if the link is stable.

**Comment by iliajie** [comment]
>> @marclaporte When configuring a Virtualmin Virtual Server, users will benefit to know the default values of the current server 

> @jcameron Linking to the docs would also be useful, if the link is stable

I will implement both .. working on it ..

**Comment by iliajie** [comment]
It will look like this, e.g.:

<img width="1180" alt="image" src="https://github.com/virtualmin/virtualmin-gpl/assets/4426533/8428078d-1190-4a0c-bd91-bcd92316ca0b">

Although, I will finish this up and make a PR only tomorrow ..

**Comment by jcameron** [comment]
Nice!

**Comment by iliajie** [comment]
Implemented in this PR https://github.com/webmin/webmin/pull/1953.
